### PR TITLE
Add ability to have Colab badges on pages

### PR DESCRIPTION
**Pull request by hamelsmu**
EDIT: this is ready for review now

This closes issue #209

### CODE REVIEW

File: nbdev/templates/jekyll.tpl [hunk 1]
```diff
@@ -7,6 +7,7 @@ sidebar: home_sidebar
 {% if resources.tags != "" and resources.tags != nil %}tags: {{resources.tags}}{% endif %}
 {% if resources.summary != "" and resources.summary != nil %}summary: "{{resources.summary}}"{% endif %}
 {% if resources.summary != "" and resources.summary != nil %}description: "{{resources.summary}}"{% endif %}
```
Comment (hamelsmu) [inline nbdev/templates/jekyll.tpl:9]: @jph00 @sgugger  I noticed this line of code while I was doing this PR.  Looks like a bug, WDYT?  should the value should be description, not the summary again, right?

**Comment by jph00** [comment]
Nice! I think we need a way to turn them on and off per page. If on for the site, it shouldn't be added to pages that aren't from notebooks.

And needs some docs including in readme

**Comment by hamelsmu** [comment]
@jph00 I made the following changes:

1. Edited `index.ipynb` to add a section about badges
2. Built the docs which rendered `README.md` as well as all of the HTML pages with the new Colab Badge
3. Added the ability to turn off a badge selectively on a page even though the site-wide option is turned off via the `hide_colab_badge` front matter, this is what is used in fastpages, BTW.
4.  Re: ignoring non-notebook posts I will comment on the code that facilitates this naturally. 

LMK what you think

File: docs/_layouts/page.html [hunk 2]
```diff
@@ -5,6 +5,9 @@
 <div class="post-header">
     <a id="{{page.title}}"></a>
     <h1 class="post-title-main">{{ page.title }}</h1>
+    {% if page.nb_path %}
```
Comment (hamelsmu) [inline docs/_layouts/page.html:8]: @jph00 this line of code prevents non-notebook documents from getting a Colab badge

**Pull request merged**